src/storage/admin/RebuildIndexTask.cpp

```diff
@@ -0,0 +1,278 @@
+                                                   operationPrefix,
+                                                   &operationIter);
+        if (operationRet != kvstore::ResultCode::SUCCEEDED) {
+            LOG(ERROR) << "Processing Part " << part << " Failed";
+            return operationRet;
+        }
+
+        std::vector<std::string> operations;
+        operations.reserve(FLAGS_rebuild_index_batch_num);
+        while (operationIter->valid()) {
+            auto opKey = operationIter->key();
+            auto opVal = operationIter->val();
+            // replay operation record
+            if (OperationKeyUtils::isModifyOperation(opKey)) {
+                VLOG(3) << "Processing Modify Operation " << opKey;
+                auto key = OperationKeyUtils::getOperationKey(opKey);
+                std::vector<kvstore::KV> pairs;
+                pairs.emplace_back(std::move(key), "");
+                auto ret = writeData(space, part, std::move(pairs));
+                if (kvstore::ResultCode::SUCCEEDED != ret) {
+                    LOG(ERROR) << "Modify Playback Failed";
+                    return ret;
+                }
+            } else if (OperationKeyUtils::isDeleteOperation(opKey)) {
+                VLOG(3) << "Processing Delete Operation " << opVal;
+                auto ret = removeData(space, part, opVal.str());
+                if (kvstore::ResultCode::SUCCEEDED != ret) {
+                    LOG(ERROR) << "Delete Playback Failed";
+                    return ret;
+                }
+            } else {
+                LOG(ERROR) << "Unknow Operation Type";
+                return kvstore::ResultCode::ERR_INVALID_OPERATION;
+            }
+
+            operations.emplace_back(opKey);
+            if (static_cast<int32_t>(operations.size()) == FLAGS_rebuild_index_batch_num) {
+                auto ret = cleanupOperationLogs(space, part, operations);
+                if (kvstore::ResultCode::SUCCEEDED != ret) {
+                    LOG(ERROR) << "Delete Operation Failed";
+                    return ret;
+                }
+
+                operations.clear();
+            }
+            operationIter->next();
+        }
+
+        auto ret = cleanupOperationLogs(space, part, operations);
+        if (kvstore::ResultCode::SUCCEEDED != ret) {
+            LOG(ERROR) << "Cleanup Operation Failed";
+            return ret;
+        }
+
+        // When the processed operation number is less than the locked threshold,
+        // we will mark the lock's building in StorageEnv and refuse writing for
+        // a short piece of time.
+        if (static_cast<int32_t>(operations.size()) < FLAGS_rebuild_index_locked_threshold) {
+            // lock the part
+            auto key = std::make_tuple(space, indexID, part);
+            auto stateIter = env_->rebuildIndexGuard_->find(key);
+            // If the state is LOCKED, we should finished the building index successful.
```

Comment:
`If the state is LOCKED, we should finished the building index successful. ` Why we don't wait until is FINISHED?

Reply:
Currently it will waiting for on flying request finished But I forget to update comment thanks